python/cudf/cudf/utils/ioutils.py

```diff
@@ -473,8 +473,36 @@
 engine : {{ 'auto', 'cudf', 'cudf_experimental', 'pandas' }}, default 'auto'
     Parser engine to use. If 'auto' is passed, the engine will be
     automatically selected based on the other parameters.
-orient : string,
-    Indication of expected JSON string format (pandas engine only).
+lines : boolean, default False
+    Read the file as a json object per line.
+dtype : boolean or dict, default True
+    If True, infer dtypes, if a dict of column to dtype, then use those,
+    if False, then don't infer dtypes at all, applies only to the data.
+byte_range : list or tuple, default None (cudf engine only)
+    Byte range within the input file to be read.
+    The first number is the offset in bytes, the second number is the range
+    size in bytes. Set the size to zero to read all data after the offset
+    location. Reads the row that starts before or at the end of the range,
+    even if it ends after the end of the range.
```

Comment:
```
Pair of `(offset, length)` specifying a subrange of the file to be read, in bytes.
To read from `offset` to the end of the file, set `length=0`. Reads the starting
before or at the end of the range even if it ends past the end of the range.
```

What does "at the end of the range" mean? I guess the byte range specifies a semi-open interval `[offset, offset+length)` does that mean if a row starts at `offset + length - 1` then we read the entire row?

Aside: this is not a very ergonomic way of specifying "read from this offset to the end of the file". Could we accept either an `offset` int or an `(offset, length)` pair?

Reply:
This too needs a bit of re-work like you said. I'll try to address this in https://github.com/rapidsai/cudf/issues/11780